### PR TITLE
[PAXWEB-1095] Upgrade to Jetty 9.4.x

### PR DESCRIPTION
**Pull request by jbonofre**
It's a WIP for now.

### CODE REVIEW

**Comment by jbonofre** [comment]
Some itests are failing, however, if you can already take a look that would be great ! Thanks !

**Comment by jbonofre** [comment]
@ANierbeck SessionManager has been removed from Jetty 9.4.x, replace by SessionHandler (with most of methods moved there). Let me check how to refactore this part.

**Comment by gnodet** [comment]
I just tried it in Karaf.  The first problem I see is that jetty provider somehow logs to stderr instead of using a pax-logging.

**Comment by jbonofre** [comment]
Let me check. Thanks for the update.

**Comment by gnodet** [comment]
The jetty classloading is broken in OSGi  because of https://github.com/eclipse/jetty.project/commit/a311c8bde171551a57ff28c04851a3b53acfc1e0

File: pax-web-itest/pax-web-itest-container/pax-web-itest-container-jetty/src/test/java/org/ops4j/pax/web/itest/jetty/WebContainerSpdyIntegrationTest.java [hunk 1]
```diff
@@ -112,6 +112,7 @@ public void tearDown() throws BundleException {
 	 * wrapped into a bundle called pax-exam-probe
 	 */
 	@Test
+	@Ignore("java.lang.ClassNotFoundException: org.eclipse.jetty.http2.api.Session$Listener not found by PAXEXAM-PROBE")
```
Comment (gnodet) [inline pax-web-itest/pax-web-itest-container/pax-web-itest-container-jetty/src/test/java/org/ops4j/pax/web/itest/jetty/WebContainerSpdyIntegrationTest.java]: I suppose this test should be fixed somehow.

File: pax-web-itest/pax-web-itest-container/pax-web-itest-container-jetty/src/test/java/org/ops4j/pax/web/itest/jetty/support/DocumentServlet.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import aQute.bnd.annotation.component.Activate;
 import aQute.bnd.annotation.component.Component;
 import org.eclipse.jetty.http.*;
+import org.eclipse.jetty.http.pathmap.MappedResource;
```
Comment (gnodet) [inline pax-web-itest/pax-web-itest-container/pax-web-itest-container-jetty/src/test/java/org/ops4j/pax/web/itest/jetty/support/DocumentServlet.java]: This import is not used

File: pax-web-itest/pax-web-itest-container/pax-web-itest-container-jetty/src/test/java/org/ops4j/pax/web/itest/jetty/WebContainerSpdyIntegrationTest.java (hunk 1 above)
Comment (jbonofre) [reply pax-web-itest/pax-web-itest-container/pax-web-itest-container-jetty/src/test/java/org/ops4j/pax/web/itest/jetty/WebContainerSpdyIntegrationTest.java]: Thanks, I gonna fix this one.

File: pax-web-itest/pax-web-itest-container/pax-web-itest-container-jetty/src/test/java/org/ops4j/pax/web/itest/jetty/support/DocumentServlet.java (hunk 2 above)
Comment (jbonofre) [reply pax-web-itest/pax-web-itest-container/pax-web-itest-container-jetty/src/test/java/org/ops4j/pax/web/itest/jetty/support/DocumentServlet.java]: Good catch, thanks, I'm fixing.

**Comment by jbonofre** [comment]
Testing a complete build locally before merging.

**Comment by jbonofre** [comment]
Hmmm it seems the Karaf itests are failing, I'm checking.

**Pull request merged**